exoscale/resource_exoscale_nlb.go

```diff
@@ -0,0 +1,317 @@
+	if err != nil {
+		return err
+	}
+
+	if nlb == nil {
+		return fmt.Errorf("Network Load Balancer %q not found", d.Id())
+	}
+
+	if d.HasChange("name") {
+		nlb.Name = d.Get("name").(string)
+	}
+
+	if d.HasChange("description") {
+		nlb.Description = d.Get("description").(string)
+	}
+
+	zone := d.Get("zone").(string)
+
+	ctx = apiv2.WithEndpoint(ctx, apiv2.NewReqEndpoint(getEnvironment(meta), zone))
+	_, err = client.UpdateNetworkLoadBalancer(ctx, zone, nlb)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] %s: update finished successfully", resourceNLBIDString(d))
+
+	return resourceNLBRead(d, meta)
+}
+
+func resourceNLBDelete(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[DEBUG] %s: beginning delete", resourceNLBIDString(d))
+
+	ctx, cancel := context.WithTimeout(context.Background(), d.Timeout(schema.TimeoutDelete))
+	defer cancel()
+
+	client := GetComputeClient(meta)
+
+	zone := d.Get("zone").(string)
+
+	ctx = apiv2.WithEndpoint(ctx, apiv2.NewReqEndpoint(getEnvironment(meta), zone))
+	err := client.DeleteNetworkLoadBalancer(ctx, zone, d.Id())
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] %s: delete finished successfully", resourceNLBIDString(d))
+
+	return nil
+}
+
+func resourceNLBImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	log.Printf("[DEBUG] %s: beginning import", resourceNLBIDString(d))
+	ctx, cancel := context.WithTimeout(context.Background(), d.Timeout(schema.TimeoutRead))
+	defer cancel()
+
+	client := GetComputeClient(meta)
+
+	nlb, err := findNLB(ctx, d, meta)
+	if err != nil {
+		return nil, err
+	}
+
+	if nlb == nil {
+		return nil, fmt.Errorf("Network Load Balancer %q not found", d.Id())
+	}
+
+	if err := resourceNLBApply(ctx, client, d, nlb); err != nil {
+		return nil, err
+	}
+
+	resources := make([]*schema.ResourceData, 0, 1+len(nlb.Services))
+	resources = append(resources, d)
+
+	for _, service := range nlb.Services {
+		resource := resourceNLBService()
+		d := resource.Data(nil)
+		d.SetType("exoscale_nlb_service")
+		err := resourceNLBServiceApply(ctx, client, d, service)
+		if err != nil {
+			return nil, err
+		}
+
+		resources = append(resources, d)
+	}
+
+	log.Printf("[DEBUG] %s: import finished successfully", resourceNLBIDString(d))
+
+	return resources, nil
+}
+
+func resourceNLBApply(ctx context.Context, client *egoscale.Client, d *schema.ResourceData, nlb *egoscale.NetworkLoadBalancer) error {
+	if err := d.Set("name", nlb.Name); err != nil {
+		return err
+	}
+
+	if err := d.Set("description", nlb.Description); err != nil {
+		return err
+	}
+
+	if err := d.Set("created_at", nlb.CreatedAt.String()); err != nil {
+		return err
+	}
+
+	if err := d.Set("state", nlb.State); err != nil {
+		return err
+	}
+
+	if err := d.Set("ip_address", nlb.IPAddress.String()); err != nil {
+		return err
+	}
+
+	services := make([]string, len(nlb.Services))
+	for i, service := range nlb.Services {
+		services[i] = service.Name
```

Comment:
I suggest exposing the service ID instead of the name, so they can be reused in another module or in outputs for additional actions (which require the ID):

```suggestion
		services[i] = service.ID
```